react-basic-practice/src/pages/ProductDetail/index.jsx

```diff
@@ -0,0 +1,166 @@
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { getProductById } from '@/services/Products';
+import { updateCart, getCartByUserId } from '@/services/Cart';
+import DefaultLayout from '@/layouts/DefaultLayout';
+import Quantity from '@/components/Quantity';
+import Button from '@/components/common/Button';
+import Toast from '@/components/Toast';
+
+import { useLoading } from '@/contexts/LoadingProvider';
+import { useToast } from '@/contexts/ToastProvider';
+
+import './index.css';
+
+const ProductDetail = () => {
+  const { productId } = useParams();
+  const { loading, setLoading } = useLoading();
+  const [product, setProduct] = useState({});
+  const [quantityProduct, setQuantity] = useState(0);
+  const { toast, setToast } = useToast();
+
+  useEffect(() => {
+    const getData = async () => {
+      try {
+        setLoading(true);
+        const data = await getProductById(productId);
+        data ? setProduct(data) : setProduct({});
+      } catch {
+        alert('Error loading data, please reload the page');
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    getData();
+  }, []);
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      setToast((preToast) => ({
```

Comment:
What is the param of setToast that you defined?

Reply:
Are you referring to the initial value of `setToas`t ? I have defined in the file `Toast`